app/src/main/java/com/keylesspalace/tusky/ComposeActivity.java

```diff
@@ -429,13 +433,19 @@ public void onFailure(@NonNull Call<List<Emoji>> call, @NonNull Throwable t) {
                 textEditor.setText(savedTootText);
             }
 
+            // try to redo a list of media
             String savedJsonUrls = intent.getStringExtra(SAVED_JSON_URLS_EXTRA);
+            String savedJsonDescriptions = intent.getStringExtra(SAVED_JSON_DESCRIPTIONS_EXTRA);
             if (!TextUtils.isEmpty(savedJsonUrls)) {
-                // try to redo a list of media
                 loadedDraftMediaUris = new Gson().fromJson(savedJsonUrls,
                         new TypeToken<ArrayList<String>>() {
                         }.getType());
             }
+            if (!TextUtils.isEmpty(savedJsonDescriptions)) {
+                loadedDraftMediaDescriptions = new Gson().fromJson(savedJsonDescriptions,
```

Comment:
I don't like the `new Gson()` because of performance. I think you should be able to inject a shared Gson instance into the activity.